__tests__/app.test.js

```diff
@@ -173,63 +174,94 @@ describe("GET /api/articles/:article_id", () => {
       });
   });
 
-  describe("GET /api/articles/:article_id/comments", () => {
-    test("GET 200 - responds with an array of comments for the given article_id of which each comment should have the relevant properties", () => {
+  describe("GET /api/articles/:article_id (comment_count)", () => {
+    test("GET 200 /api/articles/:article_id - responds an article object, which should have the relevant properties including a comment_count", () => {
+      const articleId = 9;
       return request(app)
-        .get(`/api/articles/1/comments`)
+        .get(`/api/articles/${articleId}`)
         .expect(200)
-        .then(({ body: { comments } }) => {
-          expect(Array.isArray(comments)).toBe(true);
-          expect(comments.length).toBe(11);
-          comments.forEach((comment) => {
-            expect(comment).toMatchObject({
-              comment_id: expect.any(Number),
-              votes: expect.any(Number),
-              created_at: expect.any(String),
-              author: expect.any(String),
-              body: expect.any(String),
-              article_id: 1,
-            });
+        .then(({ body: { article } }) => {
+          expect(article).toMatchObject({
+            article_id: articleId,
+            created_at: convertTimestampToDate(1591438200000),
+            comment_count: expect.any(Number),
+            comment_count: 2,
           });
         });
     });
-    test("GET 200 - responds with the most recent comments first", () => {
+    test("GET 200 /api/articles/:article_id - if comment is not yet made for referenced article", () => {
+      const articleId = 2;
       return request(app)
-        .get(`/api/articles/1/comments`)
+        .get(`/api/articles/${articleId}`)
         .expect(200)
-        .then(({ body: { comments } }) => {
-          expect(comments).toBeSortedBy("created_at", {
-            descending: true,
+        .then(({ body: { article } }) => {
+          expect(article).toMatchObject({
+            article_id: articleId,
+            created_at: convertTimestampToDate(1602828180000),
+            comment_count: expect.any(Number),
+            comment_count: 0,
           });
         });
     });
-    test("GET 404 -- responds with approriate status and message, when the request has invalid article id number", () => {
-      const articleId = 999;
-      return request(app)
-        .get(`/api/articles/${articleId}/comments`)
-        .expect(404)
-        .then(({ body: { msg } }) => {
-          expect(msg).toBe("Not Found");
-        });
-    });
-    test("GET 400 -- responds with approriate status and message, when the request has invalid article id type", () => {
-      const articleId = "papers";
-      return request(app)
-        .get(`/api/articles/${articleId}/comments`)
-        .expect(400)
-        .then(({ body: { msg } }) => {
-          expect(msg).toBe("Bad Request");
+  });
+});
+
+describe("GET /api/articles/:article_id/comments", () => {
```

Comment:
We would have wanted to create a new test not change the old one. The reason is that if a project that has been going on for several years, there are good chances that an endpoint gets reworked. Ideally the old tests should not fail and should not be changed because the new developer working on it should not go back to the old tests and find this specific one out of the 1000s written in the years by their colleagues. Hope it makes sense 🙂